msal/src/Microsoft.Identity.Client/UIParent.cs

```diff
@@ -116,20 +118,24 @@ public static bool IsSystemWebviewAvailable()
 
             int counter = 0;
 
-            ApplicationInfo applicationInfo = Application.Context.PackageManager.GetApplicationInfo(_chromePackages[0], 0);
-
-            for (int i = 0; i < _chromePackages.Length; i++)
+            ApplicationInfo applicationInfo = Application.Context.PackageManager.GetApplicationInfo(_customTabPackages[0], 0);
+            List<string> customTabPackages = new List<string>();
+            for (int i = 0; i < _customTabPackages.Length; i++)
             {
                 try
                 {
-                    packageManager.GetPackageInfo(_chromePackages[i], PackageInfoFlags.Activities);
+                    customTabPackages.Add(packageManager.GetPackageInfo(_customTabPackages[i], PackageInfoFlags.Activities).ToString());
                     counter++;
                 }
                 catch (PackageManager.NameNotFoundException)
                 {
                 }
             }
-            if (counter > 0 && applicationInfo.Enabled == true)
+            if (counter == 1 && !string.IsNullOrEmpty(customTabPackages[0]) && !applicationInfo.Enabled)
```

Comment:
If you guys have a better suggestion for this, let me know. :)
Basically, Chrome can not be uninstalled (well, it's really complicated), so most users will just disable it, but the package will still show up, but was `applicationInfo.Enabled` false, so we need to check that `com.android.chrome` is disabled, otherwise, we'll return a count of 1, but system browser will fail.

Reply:
Background info if you want it:
Native android is making the decision for the dev to switch to embedded webview if no browsers are available. Due to security issues (ex. Google auth does not accept embedded webview), we decided to let the dev choose instead of the library, which is what this helper function is for. If the dev doesn't care if embedded or system browser is used, they can run this method to check if the device can handle system browser, and if not, launches embedded, but this is the dev devision. The experience is invisible to the user.

Reply:
I suppose this is valid also if the counter is larger than 1 ? Thus it has to do this validation for the browser used

---
In reply to: [227610888](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227610888) [](ancestors = 227610888)

Reply:
You should add this line about Chrome as a comment in the code.

---
In reply to: [227610888](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227610888) [](ancestors = 227610888)

Reply:
I think i understand your question. If the counter is larger than 1, it doesn't matter because it means there is a package w/custom tabs available. so even if chrome is disabled, there is something else, and that package will be used (not chrome). does that answer your question?

---
In reply to: [227649549](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227649549) [](ancestors = 227649549,227610888)